compileoptions.cmake

```diff
@@ -23,8 +23,10 @@ if (CLR_CMAKE_PLATFORM_UNIX)
   # after hitting just about 20 errors.
   add_compile_options(-ferror-limit=4096)
 
-  # All warnings that are not explicitly disabled are reported as errors
-  add_compile_options(-Werror)
+  if (CLR_CMAKE_WARNINGS_ARE_ERRORS)
```

Comment:
@gkhanna79 this "if" doesn't check if the symbol is defined, but rather its value. Condition is true if the variable is 1, ON, YES, TRUE, Y, or a non-zero number. False if the variable is 0, OFF, NO, FALSE, N, IGNORE, NOTFOUND, the empty string, or ends in the suffix -NOTFOUND